tests/test_remo.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import argparse
+import shutil
+import sys
+import tempfile
+import unittest
+
+import httpretty
+
+if not '..' in sys.path:
+    sys.path.insert(0, '..')
+
+from perceval.cache import Cache
+from perceval.errors import CacheError
+from perceval.backends.remo import ReMo, ReMoCommand, ReMoClient
+
+
+MOZILLA_REPS_SERVER_URL = 'http://example.com'
+MOZILLA_REPS_API = MOZILLA_REPS_SERVER_URL + '/api/v1/'
+MOZILLA_REPS_API_EVENTS = MOZILLA_REPS_SERVER_URL + '/api/v1/event/'
+
+
+def read_file(filename, mode='r'):
+    with open(filename, mode) as f:
+        content = f.read()
+    return content
+
+
+class HTTPServer():
+
+    requests_http = []  # requests done to the server
+
+    @classmethod
+    def routes(cls, empty=False):
+        """Configure in http the routes to be served"""
+
+        mozilla_events_1 = read_file('data/remo_events_1_2.json')
+        mozilla_events_2 = read_file('data/remo_events_2_2.json')
+        if empty:
+            mozilla_events_1 = read_file('data/remo_events_empty.json')
+
+        def request_callback(method, uri, headers):
+            offset = uri.split("offset=")[1].split("&")[0]
+            if offset == "0":
+                body = mozilla_events_1
+            else:
+                body = mozilla_events_2
+
+            HTTPServer.requests_http.append(httpretty.last_request())
+
+            return (200, headers, body)
+
+        httpretty.register_uri(httpretty.GET,
+                               MOZILLA_REPS_API_EVENTS,
+                               responses=[
+                                    httpretty.Response(body=request_callback)
+                               ])
+
+
+class TestReMoBackend(unittest.TestCase):
+    """ReMo backend tests"""
+
+    def test_initialization(self):
+        """Test whether attributes are initializated"""
+
+        remo = ReMo(MOZILLA_REPS_SERVER_URL, origin='test')
+
+        self.assertEqual(remo.url, MOZILLA_REPS_SERVER_URL)
```

Comment:
It would be useful to check when the `url` is `None` it is changed to the default value as the documentation says.